test_libs/pyspec/eth2spec/test/block_processing/test_process_attestation.py

```diff
@@ -191,111 +196,119 @@ def test_invalid_current_source_root(state):
+    yield from run_attestation_processing(spec, state, attestation, False)
 
 
+@with_all_phases
 @spec_state_test
-def test_bad_parent_crosslink(state):
-    next_epoch(state)
-    apply_empty_block(state)
+def test_bad_parent_crosslink(spec, state):
+    next_epoch(spec, state)
+    apply_empty_block(spec, state)
 
-    attestation = get_valid_attestation(state, signed=True)
+    attestation = get_valid_attestation(spec, state, signed=True)
     for _ in range(spec.MIN_ATTESTATION_INCLUSION_DELAY):
-        next_slot(state)
-    apply_empty_block(state)
+        next_slot(spec, state)
+    apply_empty_block(spec, state)
 
     attestation.data.crosslink.parent_root = b'\x27' * 32
 
-    yield from run_attestation_processing(state, attestation, False)
+    yield from run_attestation_processing(spec, state, attestation, False)
 
 
+@with_all_phases
 @spec_state_test
-def test_bad_crosslink_start_epoch(state):
-    next_epoch(state)
-    apply_empty_block(state)
+def test_bad_crosslink_start_epoch(spec, state):
+    next_epoch(spec, state)
+    apply_empty_block(spec, state)
 
-    attestation = get_valid_attestation(state, signed=True)
+    attestation = get_valid_attestation(spec, state, signed=True)
     for _ in range(spec.MIN_ATTESTATION_INCLUSION_DELAY):
-        next_slot(state)
-    apply_empty_block(state)
+        next_slot(spec, state)
+    apply_empty_block(spec, state)
 
     attestation.data.crosslink.start_epoch += 1
 
-    yield from run_attestation_processing(state, attestation, False)
+    yield from run_attestation_processing(spec, state, attestation, False)
 
 
+@with_all_phases
 @spec_state_test
-def test_bad_crosslink_end_epoch(state):
-    next_epoch(state)
-    apply_empty_block(state)
+def test_bad_crosslink_end_epoch(spec, state):
+    next_epoch(spec, state)
+    apply_empty_block(spec, state)
 
-    attestation = get_valid_attestation(state, signed=True)
+    attestation = get_valid_attestation(spec, state, signed=True)
     for _ in range(spec.MIN_ATTESTATION_INCLUSION_DELAY):
-        next_slot(state)
-    apply_empty_block(state)
+        next_slot(spec, state)
+    apply_empty_block(spec, state)
 
     attestation.data.crosslink.end_epoch += 1
 
-    yield from run_attestation_processing(state, attestation, False)
+    yield from run_attestation_processing(spec, state, attestation, False)
 
 
+@with_all_phases
 @spec_state_test
-def test_inconsistent_bitfields(state):
-    attestation = get_valid_attestation(state)
+def test_inconsistent_bitfields(spec, state):
+    attestation = get_valid_attestation(spec, state)
     state.slot += spec.MIN_ATTESTATION_INCLUSION_DELAY
 
     attestation.custody_bitfield = deepcopy(attestation.aggregation_bitfield) + b'\x00'
 
-    sign_attestation(state, attestation)
+    sign_attestation(spec, state, attestation)
 
-    yield from run_attestation_processing(state, attestation, False)
+    yield from run_attestation_processing(spec, state, attestation, False)
 
 
+@with_all_phases
```

Comment:
This is only a phase 0 test

```suggestion
@with_phase0
```

(need to handle import as well)